_posts/2018-01-04-notification-channels-android-o.md

```diff
@@ -0,0 +1,129 @@
+
+You can specify further customizations for the channel you have created. For example, you can specify the light colors, vibration pattern etc for the channel as shown below:
+
+```kotlin
+with(privateMessagesChannel) {
+    lightColor = Color.RED
+    enableVibration(true)
+    vibrationPattern = longArrayOf(100, 200, 300, 400)
+}
+```
+
+### Sending notifications for a notification channel
+
+Now that we've created the channel, next step is to actually specify the channel whenever we want to show a notification and that is done when creating a new notification.
+Specifying a channel for the notification looks like:
+
+```kotlin
+val notification = NotificationCompat.Builder(context)
+                .setSmallIcon(R.drawable.ic_notification_small)
+                .setContentTitle(context.getString(R.string.notif_pm_title))
+                .setContentText(message)
+                /**
+                 * further notification customizations
+                 */
+                .setChannelId(PRIVATE_MESSAGES_CHANNEL_ID)
+                .build()
+        notificationManager.notify(PM_NOTIFICATION_ID, notification)
+```
+The part of the code above that sets the channel is `.setChannelId()`. Without this line, if you're targeting Android O and above, your notification will not be shown to the user.
+
+### Updating/reading notification channels settings
+
+After creating notification channels, the ability to control is left to the user. They can turn off notifications, change priority, lights & vibration settings etc for any channel they desire. However, it is possible to programmatically read the current channel settings. You may need this, so that you can present the status of the notification channel to your user - for example, if the user has turned off notification for a certain group
+chat, you may display a "mute" icon next to the group details or somewhere it's visible to the user. 
+
+It's important not to make this intrusive in anyway or obstructing them from going about their normal activities in the app because it defeats the purpose of giving users the power to control the settings for the channel.
+
+To read the notification channel details, you need to get a reference to the notification channel, and query the settings you desire.
+
+```kotlin
+val pmNotificationChannel = notificationManager.getNotificationChannel(GROUP_CHAT_CHANNEL_ID)
+
+val channelIsBlocked = pmNotificationChannel.importance == NotificationManager.IMPORTANCE_NONE
+
+if (channelIsBlocked) {
+	// do something unintrusive to make user aware that they blocked the notification
+	// you should also provide a way to make them go to the channel settings
+}
+```
+
+### Deleting notification channels
+So, assuming that a user has unsubscribed from a previous conversation that a channel was created for, and you have to delete the channel. Deleting is quite straightforward, to do this, one needs to do something like:
```

Comment:
_So, assuming that a user has unsubscribed from a previous conversation that a channel was created for, and you have to delete the channel. Deleting is quite straightforward, to do this, one needs to do something like:_

_So let's assume that a user has unsubscribed from a previous conversation that a channel was created for. You'll have to delete the channel. Deleting is quite straightforward. To do this, one needs to do something like:_

Reply:
🔒